From 0df55f80e20e1a075d487b3109016fad0b8ee124 Mon Sep 17 00:00:00 2001
From: yocto <yocto@androidbuilder>
Date: Wed, 14 Sep 2016 11:49:59 +0200
Subject: [PATCH] fix sdcard image creation to allow dtb with '.' inside
 filename

-our dtb "imx6-egf-WID0510_AA01.01.dtb" could not be found because of the
'.' in the name
---
 classes/image_types_fsl.bbclass | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/classes/image_types_fsl.bbclass b/classes/image_types_fsl.bbclass
index d351348..f7e6fca 100644
--- a/classes/image_types_fsl.bbclass
+++ b/classes/image_types_fsl.bbclass
@@ -127,7 +127,7 @@ _generate_boot_image() {
 	# Copy device tree file
 	if test -n "${KERNEL_DEVICETREE}"; then
 		for DTS_FILE in ${KERNEL_DEVICETREE}; do
-			DTS_BASE_NAME=`basename ${DTS_FILE} | awk -F "." '{print $1}'`
+			DTS_BASE_NAME=`basename ${DTS_FILE} | awk -F ".dtb" '{print $1}'`
 			if [ -e "${DEPLOY_DIR_IMAGE}/${KERNEL_IMAGETYPE}-${DTS_BASE_NAME}.dtb" ]; then
 				kernel_bin="`readlink ${DEPLOY_DIR_IMAGE}/${KERNEL_IMAGETYPE}-${MACHINE}.bin`"
 				kernel_bin_for_dtb="`readlink ${DEPLOY_DIR_IMAGE}/${KERNEL_IMAGETYPE}-${DTS_BASE_NAME}.dtb | sed "s,$DTS_BASE_NAME,${MACHINE},g;s,\.dtb$,.bin,g"`"
@@ -135,7 +135,7 @@ _generate_boot_image() {
 					mcopy -i ${WORKDIR}/boot.img -s ${DEPLOY_DIR_IMAGE}/${KERNEL_IMAGETYPE}-${DTS_BASE_NAME}.dtb ::/${DTS_BASE_NAME}.dtb
 				fi
 			else
-				bbfatal "${DTS_FILE} does not exist."
+				bbfatal "${DTS_FILE} does not exist. ${DEPLOY_DIR_IMAGE}/${KERNEL_IMAGETYPE}-${DTS_BASE_NAME}.dtb"
 			fi
 		done
 	fi
-- 
2.7.4

